kratos/solving_strategies/convergencecriterias/residual_criteria.h

```diff
@@ -359,12 +410,18 @@ class ResidualCriteria
         const TSystemVectorType& rb
         )
     {
+        // Retrieve the data communicator
+        const auto& r_data_communicator = rModelPart.GetCommunicator().GetDataCommunicator();
+
+        // The current MPI rank
+        const int rank = r_data_communicator.Rank();
+
         // Initialize
         TDataType residual_solution_norm = TDataType();
-        SizeType dof_num = 0;
+        unsigned int dof_num = 0;
```

Comment:
Why to change from `SzieType` to `unsigned int` if you are going to cast it back in 425 ?

Reply:
Because in Windows DataCOmmunicator does not work for long unisgned int

Reply:
It always have to be windows :cry:

Reply:
Windows!!!

![86wrx9](https://github.com/KratosMultiphysics/Kratos/assets/19991680/2314ae45-00de-46f1-87af-3deac38dd80b)